### PR TITLE
[DFMD-11] Sign-Up Form Validation

### PR DESCRIPTION
**Pull request by damlois**
This PR contains my implementation for sign up form validation

image
image
image
image
image

### CODE REVIEW

File: submissions/loisAdegbohungbe/DOM/signup.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  const nameRegex = /^[A-Za-z0-9 ]+$/,
+    phoneRegex = /^[\+]?[0-9]{8,14}$/i,
+    emailRegex = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,10})+$/,
+    passwordRegex = /(?=.*\d)(?=.*[a-z])(?=.*[a-zA-Z]).{6,}$/;
+  validation(
+    firstName,
+    nameRegex,
+    'Please enter your First Name',
+    'Please enter a valid First Name ',
+    event
```
Comment (ToluAkin) [inline submissions/loisAdegbohungbe/DOM/signup.js:52]: @damlois Why are you passing in event as the last parameter?

Comment (ToluAkin) [reply submissions/loisAdegbohungbe/DOM/signup.js:52]: You only need it in the function

**Review by ToluAkin — CHANGES_REQUESTED** [review]

File: submissions/loisAdegbohungbe/DOM/signup.js (hunk 1 above)
Comment (damlois) [reply submissions/loisAdegbohungbe/DOM/signup.js:52]: It won't work if not passed @ToluAkin

File: submissions/loisAdegbohungbe/DOM/signup.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    event.preventDefault();
+    return false;
+  } else {
+    field.nextElementSibling.textContent = '';
+    return true;
+  }
+}
+
+confirmPasswordCheck = (passwordField, confirmPasswordField, error1, error2,  event) => {
+    event.preventDefault();
```
Comment (ToluAkin) [inline submissions/loisAdegbohungbe/DOM/signup.js:26]: Remove `event.preventDefault();` here

File: submissions/loisAdegbohungbe/DOM/signup.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
+let  form = document.getElementById('signup-form');
+let  firstName = document.getElementById('firstName');
+let  lastName = document.getElementById('lastName');
+let  phoneNumber = document.getElementById('phone');
+let  email = document.getElementById('email');
+let  password = document.getElementById('password');
+let  confirmPassword = document.getElementById('confirmPassword');
+
+validation = (field, regExp, error1, error2, event) => {
+  event.preventDefault();
```
Comment (ToluAkin) [inline submissions/loisAdegbohungbe/DOM/signup.js:10]: Remove `event.preventDefault();` here

**Review by ToluAkin — CHANGES_REQUESTED** [review]

File: submissions/loisAdegbohungbe/DOM/signup.js (hunk 3 above)
Comment (damlois) [reply submissions/loisAdegbohungbe/DOM/signup.js]: done @ToluAkin

**Review by ToluAkin — APPROVED** [review]

**Pull request merged**